propertyestimator/backends/dask.py

```diff
@@ -1,13 +1,228 @@
+        >>>
+        >>> # Define the set of commands which will set up the correct environment
+        >>> # for each of the workers.
+        >>> worker_script_commands = [
+        >>>     'module load cuda/9.2',
+        >>> ]
+        >>>
+        >>> # Create the backend which will adaptively try to spin up between one and
+        >>> # ten workers with the requested resources depending on the calculation load.
+        >>> from propertyestimator.backends import DaskLSFBackend
+        >>> from simtk.unit import unit
+        >>> lsf_backend = DaskLSFBackend(minimum_number_of_workers=1,
+        >>>                              maximum_number_of_workers=10,
+        >>>                              resources_per_worker=resources,
+        >>>                              default_memory_unit=unit.gigabyte,
+        >>>                              queue_name='gpuqueue',
+        >>>                              extra_script_commands=worker_script_commands)
+        """
+
+        super().__init__(minimum_number_of_workers, resources_per_worker)
+
+        assert isinstance(resources_per_worker, QueueWorkerResources)
+
+        assert minimum_number_of_workers <= maximum_number_of_workers
+
+        if resources_per_worker.number_of_gpus > 0:
+
+            if resources_per_worker.preferred_gpu_toolkit == ComputeResources.GPUToolkit.OpenCL:
+                raise ValueError('The OpenCL gpu backend is not currently supported.')
+
+            if resources_per_worker.number_of_gpus > 1:
+                raise ValueError('Only one GPU per worker is currently supported.')
+
+        self._cluster = None
+        self._client = None
+
+        self._minimum_number_of_workers = minimum_number_of_workers
+        self._maximum_number_of_workers = maximum_number_of_workers
+
+        self._default_memory_unit = default_memory_unit
+
+        self._queue_name = queue_name
+
+        self._extra_script_commands = extra_script_commands
+
+        self._adaptive_interval = adaptive_interval
+
+    def start(self):
+
+        from dask_jobqueue import LSFCluster
+
+        requested_memory = self._resources_per_worker.per_thread_memory_limit
+
+        memory_default_unit = requested_memory.value_in_unit(self._default_memory_unit)
+        memory_bytes = requested_memory.value_in_unit(unit.byte)
+
+        memory_string = '{}{}'.format(memory_default_unit, self._default_memory_unit.get_symbol())
+
+        # Dask assumes we will be using mega bytes as the default unit, so we need
+        # to multiply by a corrective factor to remove this assumption.
+        lsf_byte_scale = (1 * (unit.mega * unit.byte)).value_in_unit(self._default_memory_unit)
+        memory_bytes *= lsf_byte_scale
+
+        job_extra = None
+
+        if self._resources_per_worker.number_of_gpus > 0:
+
+            job_extra = [
+                '-gpu num={}:j_exclusive=yes:mode=shared:mps=no:'.format(self._resources_per_worker.number_of_gpus)
```

Comment:
I'm not sure how `self._resources_per_worker.number_of_gpus` should be interpreted, but I want to make sure that you are aware that with IBM LSF syntax (the one installed on Lilac), `-gpu num=4` means 4 GPUs _per node_.

Reply:
> I'm not sure how `self._resources_per_worker.number_of_gpus` should be interpreted

This whole object describes the resources which will be requested per dask worker (i.e per each submission which is made to the LSF queue) such that `number_of_gpus` is the number of gpu's which will be requested per worker. It is expected that the resources requested here should fit within a single node (unfortunately there is no simple way to validate these that I can see before a worker is requested) - I'll add a note to clarify this.

> I want to make sure that you are aware that with IBM LSF syntax (the one installed on Lilac), `-gpu num=4` means 4 GPUs _per node_.

Thanks for pointing this out - this way my understanding also!